src/OpenSpaceToolkit/Astrodynamics/NumericalSolver.cpp

```diff
@@ -0,0 +1,407 @@
+        {
+            switch (stepperType_)
+                {
+                    case NumericalSolver::StepperType::RungeKuttaCashKarp54:
+                        integrate_adaptive (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_54() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep) ;
+                        return aStateVector ;
+                    case NumericalSolver::StepperType::RungeKuttaFehlberg78:
+                        integrate_adaptive (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_78() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep) ;
+                        return aStateVector ;
+                }
+        }
+        case NumericalSolver::LogType::LogConstant:
+        {
+            switch (stepperType_)
+                {
+                    case NumericalSolver::StepperType::RungeKuttaCashKarp54:
+                        integrate_const (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_54() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep, NumericalSolver::NumericalIntegrationLogger) ;
+                        return aStateVector ;
+                    case NumericalSolver::StepperType::RungeKuttaFehlberg78:
+                        integrate_const (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_78() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep, NumericalSolver::NumericalIntegrationLogger) ;
+                        return aStateVector ;
+                }
+        }
+        case NumericalSolver::LogType::LogAdaptive:
+        {
+            switch (stepperType_)
+                {
+                    case NumericalSolver::StepperType::RungeKuttaCashKarp54:
+                        integrate_adaptive (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_54() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep, NumericalSolver::NumericalIntegrationLogger) ;
+                        return aStateVector ;
+                    case NumericalSolver::StepperType::RungeKuttaFehlberg78:
+                        integrate_adaptive (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_78() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep, NumericalSolver::NumericalIntegrationLogger) ;
+                        return aStateVector ;
+                }
+        }
+    }
+
+    throw ostk::core::error::runtime::Undefined("No StateVector returned from Odeint") ;
+
+    return NumericalSolver::StateVector(anInitialStateVector.size()) ;
+
+}
+
+
+NumericalSolver::StateVector    NumericalSolver::integrateStateFromInstantToInstant (   const   StateVector&        anInitialStateVector,
```

Comment:
Why not calling `integrateStateForDuration` under the hood to keep the code DRY?

Reply:
Ah that's a good point